pkg/testingutil/routes.go

```diff
@@ -31,6 +32,9 @@ func GetSingleRoute(urlName string, port int, componentName, applicationName str
 				applabels.OdoVersion:           version.VERSION,
 				labels.URLLabel:                urlName,
 			},
+			Annotations: map[string]string{
+				"openshift.io/host.generated": "true",
+			},
```

Comment:
We don't need this. For OpenShift clusters, creation of an ingress also creates a route. So the list of routes, also contains the routes belonging to these ingresses. But for these routes, the owner reference is set to their belonging ingresses. So we can use the owner reference to distinguish these routes. We are already doing this in the ListPushed function
https://github.com/openshift/odo/blob/8faf7e5c998344938524ef6970d3dbe3bec58c6f/pkg/url/url.go#L357

Reply:
You are right, I checked the route created for ingress, the `OwnerReference[0].Kind` is set to `Ingress`. I have updated the PR to use the OwnerReference instead of the annotation.

Reply:
OK :+1: